# Resumen

Actualmente, **sin criterios de racionalizacion** tendriamos que cerrar una brecha de **5,743 plazas** y costaria **S/ 89,832,006**. Sin embargo si se usan los siguientes **criterios de ordenamiento**:

1) Nombrados excedentes se reubican dentro de su UGEL

2) Las plazas no nombradas excedentes se reubican a nivel nacional para cubrir los requerimientos

La brecha seria de **2,976 plazas** con un costo **S/46,550,592** lo que implica un **ahorro** de **2,767 plazas** equivalentes a **S/ 43,281,414**

In [1]:
#Libraries
import os 
import pandas as pd
import numpy as np

os.chdir('D:\Brecha-no-docente') # Directorio de trabajo

In [2]:
df = pd.read_stata('Resultados\Brecha con costo.dta')

C:\Users\USUARIO\Anaconda3\lib\site-packages\pandas\io\stata.py:1457: UnicodeWarning: 
One or more strings in the dta file could not be decoded using utf-8, and
so the fallback encoding of latin-1 is being used.  This can happen when a file
has been incorrectly encoded by Stata or some other software. You should verify
the string values returned are correct.
  warnings.warn(msg, UnicodeWarning)


In [3]:
identificacion = ['d_dpto', 'region', 'pliego', 'codue', 'unidadejecutora', 'codooii', 'nombentidad', 'tipo_entidad']
datos = ['pers_limp_mant_n', 'pers_limp_mant_c', 'pers_limp_mant', 'opt_pers_limp_mant', 'exd_pers_limp_mant', 'req_pers_limp_mant',
        'pers_vigilancia_n', 'pers_vigilancia_c', 'pers_vigilancia', 'opt_pers_vigilancia', 'exd_pers_vigilancia', 'req_pers_vigilancia', 
         'coord_adm_ie_n', 'coord_adm_ie_c', 'coord_adm_ie', 'opt_coord_adm_ie', 'exd_coord_adm_ie', 'req_coord_adm_ie',
         'oficinista_n', 'oficinista_c', 'oficinista', 'opt_oficinista', 'exd_oficinista', 'req_oficinista',
         'secretario_n', 'secretario_c', 'secretario', 'opt_secretario', 'exd_secretario', 'req_secretario'
        ]

In [4]:
df_ugel = df.groupby(identificacion)[datos].sum()
df_ugel.reset_index(inplace = True)

In [5]:
personal = ['pers_limp_mant','pers_vigilancia', 'coord_adm_ie', 'oficinista', 'secretario']

# Desagrego el personal excedente
for x in personal:
    df_ugel[f'exd_{x}_c'] = df_ugel[[f'exd_{x}',f'{x}_c']].min(axis=1)  
    df_ugel[f'exd_{x}_n'] = df_ugel[f'exd_{x}'] - df_ugel[f'exd_{x}_c']

# Brecha sin racionalizacion

En esta brecha se financia todos los requerimientos

In [45]:
brecha_sr = df_ugel.copy()
for x in personal:
    brecha_sr[f'brecha_sr_{x}'] = brecha_sr[f'req_{x}']

costo_oficinista = 12*1200+600+12*(1200*0.09)
costo_secretario = 12*1400+600+12*(1400*0.09)
costo_otros      = 12*1150+600+12*(1150*0.09)

brecha_sr['costo_brecha_sr_{x}'] = brecha_sr['brecha_sin_racio']*(12*1150+600+12*(1150*0.09))

In [6]:
brecha_sr = df_ugel.copy()
brecha_sr['brecha_sin_racio'] = brecha_sr['req_pers_limp_mant']
brecha_sr['costo_brecha_sin_racio'] = brecha_sr['brecha_sin_racio']*(12*1150+600+12*(1150*0.09))

brecha_sr.rename(columns={"pers_limp_mant_n": "Nombrados", 
                   "pers_limp_mant_c": "Contratados",
                    'pers_limp_mant' : 'Personal actual',
                   "opt_pers_limp_mant": "Personal optimo",
                   "exd_pers_limp_mant_n": "Excedentes nombrados",
                   "exd_pers_limp_mant_c": "Excedentes contratados",
                   'exd_pers_limp_mant' : 'Excedentes',
                   "req_pers_limp_mant": "Requerimientos",
                   "brecha_sin_racio": "Brecha sin racio",
                   'costo_brecha_sin_racio': "Costo sin racio"
                  }, inplace=True)

# datos = ['Nombrados', 'Contratados', 'Personal optimo', 'Excedentes nombrados', 'Excedentes contratados', 'Requerimientos', 'Brecha sin racio']
# brecha_sr.to_excel( 'Resultados\Personal de limpieza.xlsx', sheet_name='UGEL sin racio')

# pd.DataFrame(brecha_sr.loc[df_ugel['d_dpto']== 'TACNA' ,['region','nombentidad','exd_pers_limp_mant', 'req_pers_limp_mant','brecha_sin_racio']]) 

In [7]:
region_sr = brecha_sr.groupby('d_dpto').sum()
region_sr

Nombrados  Contratados  Personal actual  Personal optimo  \
d_dpto                                                                    
AMAZONAS           296.0         30.0            326.0            322.0   
ANCASH             969.0        143.0           1112.0            871.0   
APURIMAC           331.0         28.0            359.0            316.0   
AREQUIPA           709.0        141.0            850.0            783.0   
AYACUCHO           396.0         66.0            462.0            464.0   
CAJAMARCA          631.0        120.0            751.0            923.0   
CALLAO             602.0        163.0            765.0            731.0   
CUSCO              625.0        119.0            744.0            955.0   
HUANCAVELICA       165.0         53.0            218.0            198.0   
HUANUCO            429.0         95.0            524.0            695.0   
ICA                605.0         98.0            703.0            807.0   
JUNIN              636.0        135.0            771.0            880.0   
LA LIBERTAD        892.0        186.0           1078.0           1520.0   
LAMBAYEQUE         537.0         86.0            623.0           1045.0   
LIMA              5049.0       1313.0           6362.0           6200.0   
LORETO             480.0        475.0            955.0           1132.0   
MADRE DE DIOS       49.0         16.0             65.0            202.0   
MOQUEGUA           154.0         34.0            188.0            136.0   
PASCO              144.0         22.0            166.0            196.0   
PIURA             1056.0        201.0           1257.0           1797.0   
PUNO               791.0         66.0            857.0            765.0   
SAN MARTIN         714.0        104.0            818.0           1009.0   
TACNA              262.0         30.0            292.0            248.0   
TUMBES             333.0         21.0            354.0            264.0   
UCAYALI            292.0         54.0            346.0            660.0   

               Excedentes  Requerimientos  Excedentes contratados  \
d_dpto                                                              
AMAZONAS             75.0            71.0                    26.0   
ANCASH              327.0            86.0                   136.0   
APURIMAC             94.0            51.0                    23.0   
AREQUIPA            194.0           127.0                   131.0   
AYACUCHO             68.0            70.0                    42.0   
CAJAMARCA            90.0           262.0                    77.0   
CALLAO              113.0            79.0                   113.0   
CUSCO                58.0           269.0                    45.0   
HUANCAVELICA         42.0            22.0                    37.0   
HUANUCO              58.0           229.0                    42.0   
ICA                  64.0           168.0                    60.0   
JUNIN                89.0           198.0                    77.0   
LA LIBERTAD         193.0           635.0                   107.0   
LAMBAYEQUE           43.0           465.0                    43.0   
LIMA               1135.0           973.0                   888.0   
LORETO              209.0           386.0                   200.0   
MADRE DE DIOS         1.0           138.0                     1.0   
MOQUEGUA             57.0             5.0                    30.0   
PASCO                22.0            52.0                    10.0   
PIURA               140.0           680.0                   138.0   
PUNO                206.0           114.0                    58.0   
SAN MARTIN          109.0           300.0                    89.0   
TACNA                54.0            10.0                    29.0   
TUMBES              112.0            22.0                    21.0   
UCAYALI              17.0           331.0                    17.0   

               Excedentes nombrados  Brecha sin racio  Costo sin racio  
d_dpto                                              

In [8]:
b_sr = brecha_sr['Brecha sin racio'].sum()
costo_sr = brecha_sr['Costo sin racio'].sum()
print(f'Financiar esta brecha de {b_sr:,} plazas costaria S/{costo_sr:,}')

Financiar esta brecha de 5,743.0 plazas costaria S/89,832,008.0


# Brecha con ordenamiento nacional

Criterios de ordenamiento:
1) Los **nombrados excedentes se reubican dentro de su propia UGEL a una IIEE con requerimiento:** El argumento de este movimiento es que al ser IIEE con mas de 140 alumnos son similares en sus caracteristicas

2) **Si aun existen nombrados excedentes ya no se reubican a una IIEE con requerimiento:** Esto se explica porque realizar un moviento dentro de la misma region puede ocasionar gastos de transporte asociados

3) **Las plazas contratadas o vacantes excedentes se mueven a nivel nacional una IIEE con requerimiento**

In [9]:
brecha_cr = df_ugel.copy()

# Brecha UGEL
brecha_cr['brecha_ugel'] = brecha_cr['exd_pers_limp_mant_n'] + brecha_cr['exd_pers_limp_mant_c'] - brecha_cr['req_pers_limp_mant']

# Nombrados que no pueden ser reasignados
brecha_cr['nom_exd_rig'] = brecha_cr['exd_pers_limp_mant_n'] - brecha_cr['req_pers_limp_mant'] 
brecha_cr['nom_exd_rig'] = np.where(brecha_cr['nom_exd_rig']<0, 0 , brecha_cr['nom_exd_rig'] )

brecha_cr.rename(columns={"pers_limp_mant_n": "Nombrados", 
                   "pers_limp_mant_c": "Contratados",
                    'pers_limp_mant' : 'Personal actual',
                   "opt_pers_limp_mant": "Personal optimo",
                   "exd_pers_limp_mant_n": "Excedentes nombrados",
                   "exd_pers_limp_mant_c": "Excedentes contratados",
                   'exd_pers_limp_mant' : 'Excedentes',
                   "req_pers_limp_mant": "Requerimientos",
                   'nom_exd_rig': 'Nombrados excedentes sin reubicar',
                   "brecha_ugel": "Brecha UGEL con racio",
                  }, inplace=True)

# #Brecha regional
brecha_cr['req1'] = brecha_cr['Requerimientos'] - brecha_cr['Excedentes nombrados'] 
brecha_cr['req1'] = np.where(brecha_cr['req1']<0,0,brecha_cr['req1'])
brecha_cr['Brecha con Racio'] = brecha_cr['Excedentes contratados'] - brecha_cr['req1']


In [10]:
# #Brecha regional
region_cr = brecha_cr.groupby('d_dpto').sum()
region_cr.drop(columns = ['req1','Brecha UGEL con racio'], inplace =  True)
region_cr

Nombrados  Contratados  Personal actual  Personal optimo  \
d_dpto                                                                    
AMAZONAS           296.0         30.0            326.0            322.0   
ANCASH             969.0        143.0           1112.0            871.0   
APURIMAC           331.0         28.0            359.0            316.0   
AREQUIPA           709.0        141.0            850.0            783.0   
AYACUCHO           396.0         66.0            462.0            464.0   
CAJAMARCA          631.0        120.0            751.0            923.0   
CALLAO             602.0        163.0            765.0            731.0   
CUSCO              625.0        119.0            744.0            955.0   
HUANCAVELICA       165.0         53.0            218.0            198.0   
HUANUCO            429.0         95.0            524.0            695.0   
ICA                605.0         98.0            703.0            807.0   
JUNIN              636.0        135.0            771.0            880.0   
LA LIBERTAD        892.0        186.0           1078.0           1520.0   
LAMBAYEQUE         537.0         86.0            623.0           1045.0   
LIMA              5049.0       1313.0           6362.0           6200.0   
LORETO             480.0        475.0            955.0           1132.0   
MADRE DE DIOS       49.0         16.0             65.0            202.0   
MOQUEGUA           154.0         34.0            188.0            136.0   
PASCO              144.0         22.0            166.0            196.0   
PIURA             1056.0        201.0           1257.0           1797.0   
PUNO               791.0         66.0            857.0            765.0   
SAN MARTIN         714.0        104.0            818.0           1009.0   
TACNA              262.0         30.0            292.0            248.0   
TUMBES             333.0         21.0            354.0            264.0   
UCAYALI            292.0         54.0            346.0            660.0   

               Excedentes  Requerimientos  Excedentes contratados  \
d_dpto                                                              
AMAZONAS             75.0            71.0                    26.0   
ANCASH              327.0            86.0                   136.0   
APURIMAC             94.0            51.0                    23.0   
AREQUIPA            194.0           127.0                   131.0   
AYACUCHO             68.0            70.0                    42.0   
CAJAMARCA            90.0           262.0                    77.0   
CALLAO              113.0            79.0                   113.0   
CUSCO                58.0           269.0                    45.0   
HUANCAVELICA         42.0            22.0                    37.0   
HUANUCO              58.0           229.0                    42.0   
ICA                  64.0           168.0                    60.0   
JUNIN                89.0           198.0                    77.0   
LA LIBERTAD         193.0           635.0                   107.0   
LAMBAYEQUE           43.0           465.0                    43.0   
LIMA               1135.0           973.0                   888.0   
LORETO              209.0           386.0                   200.0   
MADRE DE DIOS         1.0           138.0                     1.0   
MOQUEGUA             57.0             5.0                    30.0   
PASCO                22.0            52.0                    10.0   
PIURA               140.0           680.0                   138.0   
PUNO                206.0           114.0                    58.0   
SAN MARTIN          109.0           300.0                    89.0   
TACNA                54.0            10.0                    29.0   
TUMBES              112.0            22.0                    21.0   
UCAYALI              17.0           331.0                    17.0   

               Excedentes nombrados  Nombrados excedentes sin reubicar  \
d_dpto                                             

In [11]:
b_cr = np.absolute(region_cr['Brecha con Racio'].sum())
costo_cr = b_cr*(12*1150+600+12*(1150*0.09))

print(f'Financiar esta brecha de {b_cr:,} plazas costaria S/{costo_cr:,}')

Financiar esta brecha de 2,976.0 plazas costaria S/46,550,592.0


In [12]:
brecha_cr.drop(columns = ['req1', 'Brecha con Racio'], inplace =  True)

# Create a Pandas Excel writer using XlsxWriter as the engine.
writer = pd.ExcelWriter('Resultados\Personal de limpieza.xlsx', engine='xlsxwriter')

# Write each dataframe to a different worksheet.
brecha_sr.to_excel(writer, sheet_name='UGEL sin racio')
region_sr.to_excel(writer, sheet_name='Region sin racio')

brecha_cr.to_excel(writer, sheet_name='UGEL con racio')
region_cr.to_excel(writer, sheet_name='Region con racio')

# Close the Pandas Excel writer and output the Excel file.
writer.save()